test/distributed/org/apache/cassandra/distributed/upgrade/ReplicationUpgradeTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.upgrade;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+
+import org.apache.cassandra.db.DecoratedKey;
+import org.apache.cassandra.dht.Murmur3Partitioner;
+import org.apache.cassandra.distributed.UpgradeableCluster;
+import org.apache.cassandra.distributed.api.ConsistencyLevel;
+import org.apache.cassandra.distributed.shared.Versions;
+import org.apache.cassandra.utils.ByteBufferUtil;
+
+import static org.apache.cassandra.distributed.shared.AssertUtils.assertRows;
+import static org.apache.cassandra.distributed.shared.AssertUtils.row;
+
+public class ReplicationUpgradeTest extends UpgradeTestBase
+{
+    @Test
+    public void testSimpleStrategy() throws Throwable
+    {
+        new TestCase()
+        .nodes(3)
+        .nodesToUpgrade(1, 2)
+        .upgrade(Versions.Major.v22, Versions.Major.v30)
+        .upgrade(Versions.Major.v3X, Versions.Major.v4)
+        .upgrade(Versions.Major.v30, Versions.Major.v4)
+        .setup(cluster -> {
+            cluster.schemaChange("CREATE KEYSPACE test_simple WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 2};");
+            cluster.schemaChange("CREATE TABLE test_simple.names (key int PRIMARY KEY, name text)");
+        })
+        .runAfterNodeUpgrade((cluster, node) -> {
+            verifyReplication(cluster, 3);
+        })
+        .run();
+    }
+
+    // TODO: Generalize this for other replication factors?
+    private void verifyReplication(UpgradeableCluster cluster, int nodes)
+    {
+        String INSERT = "INSERT INTO test_simple.names (key, name) VALUES (?, ?)";
+        String SELECT = "SELECT * FROM test_simple.names WHERE key = ?";
+
+        List<Long> initialTokens = new ArrayList<>(nodes + 1);
+        initialTokens.add(null);
+
+        for (int i = 1; i <= nodes; i++)
+        {
+            initialTokens.add(Long.valueOf(cluster.get(i).config().get("initial_token").toString()));
+        }
+
+        // TODO: Pick some random keys to cover all three nodes as primaries.
```

Comment:
Instead of random we can use a bunch of increasing int keys, say one to ten (or more). The murmur hash makes the node selection virtually random, so with a few keys is almost sure that we are going to cover all the nodes.